backend/dataall/modules/redshift_datasets_shares/services/redshift_table_share_processor.py

```diff
@@ -18,11 +33,477 @@ def __init__(self, session, share_data, shareable_items, reapply=False):
         self.tables: List[RedshiftTable] = shareable_items
         self.reapply: bool = reapply
 
+        dataset_connection = RedshiftConnectionRepository.get_redshift_connection(
+            self.session, self.dataset.connectionUri
+        )
+        self.source_connection = dataset_connection
```

Comment:
nit:  no need for `dataset_connection` variable
```
self.source_connection = RedshiftConnectionRepository.get_redshift_connection(
    self.session, self.dataset.connectionUri
)
```

Reply:
Fixed!